src/spec.c

```diff
@@ -1054,6 +1054,7 @@ IndexSpec *IndexSpec_Parse(const char *name, const char **argv, int argc, QueryE
   return spec;
 
 failure:  // on failure free the spec fields array and return an error
+  spec->flags &= ~Index_Temporary;
```

Comment:
Good enough solution for now but I believe we should cobsidet if free the temporary index on another thread is still needed. We can do it on another PR on master only (without cherrypick)